src/gluonts/representation/discrete_pit.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from .representation import Representation
+
+# Standard library imports
+from typing import Tuple, Optional, Union, List
+import numpy as np
+import mxnet as mx
+from mxnet.gluon import nn
+
+# First-party imports
+from gluonts.core.component import validated, get_mxnet_context
+from gluonts.model.common import Tensor
+from gluonts.dataset.common import Dataset
+
+
+# LearnedBinning = Union[GlobalRelativeBinning, LocalAbsoluteBinning]
+
+
+class DiscretePIT(Representation):
+    """
+    A class representing a discrete probability integral transform of a given quantile-based learned binning. 
+    Note that this representation is intended to be applied on top of a quantile-based binning representation.
+
+    Parameters
+    ----------
+    num_bins
+        Number of bins used by the data on which this representation is applied.
+    mlp_tranf
+        Whether we want to post-process the pit-transformed valued using a MLP which can learn an appropriate
+        binning, which would ensure that pit models have the same expressiveness as standard quantile binning with
+        embedding.
+        (default: False)
+    embedding_size
+        The desired layer output size if mlp_tranf=True.
+        (default: round(num_bins**(1/4)))
+    """
+
+    @validated()
+    def __init__(
+        self,
+        num_bins: int,
+        mlp_transf: bool = False,
+        embedding_size: Optional[int] = None,
+        *args,
+        **kwargs,
+    ):
+        super().__init__(*args, **kwargs)
+
+        self.num_bins = num_bins
+        self.mlp_transf = mlp_transf
+
+        if embedding_size is None:
+            # Embedding size heuristic that seems to work well in practice. For reference see:
+            # https://developers.googleblog.com/2017/11/introducing-tensorflow-feature-columns.html
```

Comment:
Maybe this reference can go in the docstring directly, so that it's more exposed